woocommerce/woocommerce
# Remove address indexes from list of internal keys.

### All Submissions:






### Changes proposed in this Pull Request:



In #35121 we started adding _billing_address_index and _shipping_address_index to our metadata as well. However, [as pointed out here](https://github.com/woocommerce/woocommerce/pull/35121#pullrequestreview-1146604012), this causes duplicate entries for these indexes the metadata.

The reason this happens is that when checking whether metadata already exists, we filter out all metadata that is declared internal. Therefore we don't realize that metadata already exists and add another entry.

This PR fixes that issue allowing address indexes from not being filtered out.






### How to test the changes in this Pull Request:



1. Go to the order edit screen and update any address-related field.
2. Update it again, and inspect the database. You will see that the index fields are duplicated.
3. Check out this branch and try updating the order address again; the field will be updated not instead of duplicated.



### Other information:




### FOR PR REVIEWER ONLY:
